crates/xline-client/src/clients/kv.rs

```diff
@@ -173,75 +189,117 @@ impl KvClient {
-    ///         .and_then(
-    ///             &[TxnOp::put(
-    ///                 PutRequest::new("key2", "value3").with_prev_kv(true),
-    ///             )][..],
-    ///         )
-    ///         .or_else(&[TxnOp::range(RangeRequest::new("key2"))][..]);
+    ///     let resp_put = client.put("key", "val").await?;
+    ///     let rev = resp_put.header.unwrap().revision;
     ///
-    ///     let _resp = client.txn(txn_req).await?;
+    ///     let _resp = client.compact(CompactionRequest::new(rev)).await?;
     ///
     ///     Ok(())
     /// }
     /// ```
     #[inline]
-    pub async fn txn(&self, request: TxnRequest) -> Result<TxnResponse> {
-        let request = RequestWrapper::from(xlineapi::TxnRequest::from(request));
+    pub async fn compact(&self, request: CompactionRequest) -> Result<CompactionResponse> {
+        if request.physical() {
+            let mut kv_client = self.kv_client.clone();
+            return kv_client
+                .compact(xlineapi::CompactionRequest::from(request))
+                .await
+                .map(tonic::Response::into_inner)
+                .map_err(Into::into);
+        }
+        let request = RequestWrapper::from(xlineapi::CompactionRequest::from(request));
         let cmd = Command::new(request);
-        let (cmd_res, Some(sync_res)) = self
+        let (cmd_res, _sync_res) = self
             .curp_client
-            .propose(&cmd, self.token.as_ref(), false)
-            .await??
-        else {
-            unreachable!("sync_res is always Some when use_fast_path is false");
-        };
-        let mut res_wrapper = cmd_res.into_inner();
-        res_wrapper.update_revision(sync_res.revision());
-        Ok(res_wrapper.into())
+            .propose(&cmd, self.token.as_ref(), true)
+            .await??;
+        Ok(cmd_res.into_inner().into())
     }
 
-    /// Compacts the key-value store up to a given revision.
-    /// All keys with revisions less than the given revision will be compacted.
-    /// The compaction process will remove all historical versions of these keys, except for the most recent one.
-    /// For example, here is a revision list: [(A, 1), (A, 2), (A, 3), (A, 4), (A, 5)].
-    /// We compact at revision 3. After the compaction, the revision list will become [(A, 3), (A, 4), (A, 5)].
-    /// All revisions less than 3 are deleted. The latest revision, 3, will be kept.
+    // region transaction
+
+    /// Append a condition to the transaction `compare`
+    #[inline]
+    pub fn when(&mut self, compares: impl Into<Compare>) -> &mut Self {
+        self.txn.when(compares.into());
+        self
+    }
+    /// Append an operation to the transaction `success`
+    #[inline]
+    pub fn and_then<F, O>(&mut self, operation: F) -> &mut Self
+    where
+        F: FnOnce(&mut Self) -> O,
+        O: Into<TxnOp>,
+    {
+        let temp = operation(self).into();
+        self.txn.and_then(temp);
+        self
+    }
```

Comment:
I believe it should be able to accept multiple operations because it will be hard to use if the user needs to perform multiple operations.